lua/nvim_lsp/angular.lua

```diff
@@ -0,0 +1,72 @@
+local configs = require 'nvim_lsp/configs'
+local util = require 'nvim_lsp/util'
+
+local server_name = 'angularls'
+local bin_name = server_name
+local install_loc = util.base_install_dir .. '/' .. server_name
+local script_loc = install_loc .. '/node_modules/@angular/language-server/index.js'
+local bin_loc = install_loc .. '/node_modules/.bin/angularls'
+
+local installer = util.npm_installer {
+  server_name = server_name,
+  packages = { '@angular/language-server' },
+  binaries = { bin_name },
+  -- angular-language-service doesn't expose a binary, so we create an execution wrapper.
+  post_install_script =
+    'echo "#! /bin/sh\n' .. 'node ' .. script_loc .. ' \\$*' .. '" > ' .. bin_loc .. '\n' ..
+    'chmod +x ' .. bin_loc
+}
+
+local function get_probe_dir()
+  local project_root = util.find_node_modules_ancestor(vim.fn.getcwd())
+
+  return project_root and (project_root .. '/node_modules') or ''
+end
+
+local default_prob_dir = get_probe_dir()
+
+configs[server_name] = {
+  name = 'angular',
+  default_config = {
+    cmd = {
+      bin_loc,
+      '--stdio',
+      '--tsProbeLocations', default_prob_dir,
+      '--ngProbeLocations', default_prob_dir
+    },
+    filetypes = {'typescript', 'html', 'typescriptreact', 'typescript.tsx'},
+    root_dir = util.root_pattern('angular.json', 'tsconfig.json', 'package.json', '.git'),
+    on_new_config = function(new_config)
+      local install_info = installer.info()
+      local new_prob_dir = get_probe_dir()
```

Comment:
@h-michael Do you know if there is a way to get the new resolved root directory in the `on_new_config` hook? I think using the `vim.fn.getcwd()` isn't quite the right way here.